src/stdlibs.jl

```diff
@@ -0,0 +1,10 @@
+using Random: Random
+
+if isdefined(Base, :Memory)
```

Comment:
Maybe rather something like
```julia
for storage in (Vector, optional_memory...)
    @eval begin
        Random.rand!(rng::Random.AbstractRNG, A::FixedSizeArray{T,N,($storage){T}}, sp::Random.Sampler) where {N,T} = Random.rand!(rng, A.mem, sp)
        [...]
    end
end
```

Although I don't see much wrong with this either?
```julia
Random.rand!(rng::Random.AbstractRNG, A::FixedSizeArray, sp::Random.Sampler) = Random.rand!(rng, A.mem, sp)
```

Reply:
The point being it's nicer not to hardcode `Memory`.